### PR TITLE
Fix issue with not highlighted snippets and automatically scroll to snippets

### PR DESCRIPTION
**Pull request by FrostyX**
Two unrelated changes in this PR:

- Fix for #166 
- Automatical scroll to selected snippets

### CODE REVIEW

**Comment by FrostyX** [comment]
Okay, this seems to work great for about half of the snippets. For the other half, it either scrolls a little too much or not enough.

**Comment by xsuchy** [comment]
Hmm, when I try it in docker-compose, it tells me I have no data. :(
But just by reading the code it LGTM.

**Comment by FrostyX** [comment]
Do not merge yet though. There must be some bug in the calculation of how many lines should be scrolled. If we want ship the fix for not highlighted snippets, I'd rather split it into two PRs.

File: frontend/src/app/components/snippets.cljs [hunk 1]
```diff
@@ -147,3 +147,20 @@
...
+                (subs x 0 (:start-index snippet))
+                (filter #{"\n"} x)
+                (count x))
+
+        ;; I experimentally figured out this number but the the applicable CSS
+        ;; line-height is `--bs-body-line-height' which is 1.5 and the
+        ;; font-size is 0.875em which equals to 14px. The `14 * 1.5' number
+        ;; would scroll good enough but when adding 0.3, it positions the
+        ;; snippet more close to the center.
+        lineheight 21.3
```
Comment (TomasTomecek) [inline frontend/src/app/components/snippets.cljs:164]: This is the reason why webdev is so hard :D

**Review by TomasTomecek — APPROVED** [review]
I tried about 7 different reviews and in most of them it worked so well!

**Comment by FrostyX** [comment]
> Okay, this seems to work great for about half of the snippets. For the other half, it either scrolls a little too much or not enough.

I found and fixed the issue, so scrolling to snippets now works in 99% of cases. There are still two special cases that don't work:

- We only scroll vertically, horizontal scroll isn't implemented. So if a snippet is only a few words, not the whole line, and is so far to the right that we need to scroll horizontally, user needs to do the horizontal scroll manually
- When a snippet is not in the first file, we automatically switch tab to the correct file and scroll. But when user clicks on the snippet in the right column and the file needs to be switched, we switch it but we don't automatically scroll

I'd prefer merging now and fixing these corner cases in a followup PR

**Comment by FrostyX** [comment]
I don't understand why the pre-commit fails because it doesn't show any error:

```
linting took 91ms, errors: 0, warnings: 0

Lint Clojure.............................................................Passed
Error: Process completed with exit code 1.
```

**Pull request merged**